contentful/template-ecommerce-webapp-nextjs
# feat: [PLATO-421] add tracking events

**_What will change?_**

This work adds the Segment tracking events described in the [tracking plan](https://contentful.atlassian.net/wiki/spaces/PROD/pages/4138500311/Templates+-+Segment+tracking#Tracking-plan).

Ticket: https://contentful.atlassian.net/browse/PLATO-421 - Part of https://contentful.atlassian.net/browse/PLATO-411

**_Testing steps_**

- Visit the Vercel preview URL
- In the footer, click on `Manage cookie consent` and make sure Analytics cookies are accepted and AdBlockers are disabled (confirm there are no errors being thrown in the browser's console)
- Make sure all events mentioned in the tracking plan, are correctly implemented and fired accordingly. This can be checked with the Segment Debugger, available in our staging environment. Credentials can be found on 1Password.

**Example of homepage**

<img width="345" alt="ecommerce-homepage" src="https://user-images.githubusercontent.com/103024358/220393536-67bf66c3-61b5-44ec-8de6-b8d76a4a612f.png">

**Example of navigation to the black dress page**

<img width="413" alt="ecommerce-black-dress" src="https://user-images.githubusercontent.com/103024358/220393531-3f41cf0f-db4b-40e2-b03b-536be4917667.png">

- We are aware that additional page events are fired when URL parameters change (this happens for the preview and xray mode). This is an accepted side effect.

**_TODO_**
- [ ] Check the events on Segment production after the merge (sanity check, it should have been taken care of for all templates).
- [ ] Check potential script/cookies to be added to the Osano configuration - taken care of in https://contentful.atlassian.net/browse/PLATO-467.
